commit 36cc908ed549389713955093bbfeaa35fdaf3e2e
Author: Mahesh Bodapati <mahesh.bodapati@ibm.com>
Date:   Fri May 12 05:22:59 2023 -0500

    powerpc:GCC(<10) doesn't allow -mlong-double-64 after -mabi=ieeelongdouble
    
    Removed -mabi=ieeelongdouble on failing tests. It resolves the error.
    error: ‘-mabi=ieeelongdouble’ requires ‘-mlong-double-128’

diff --git a/sysdeps/powerpc/powerpc64/le/Makefile b/sysdeps/powerpc/powerpc64/le/Makefile
index 53644d50cc..5214eb40ad 100644
--- a/sysdeps/powerpc/powerpc64/le/Makefile
+++ b/sysdeps/powerpc/powerpc64/le/Makefile
@@ -158,6 +158,23 @@ ifeq ($(subdir),stdio-common)
 CFLAGS-printf_fp.c = $(type-float128-CFLAGS)
 CFLAGS-printf_fphex.c = $(type-float128-CFLAGS)
 CFLAGS-printf_size.c = $(type-float128-CFLAGS)
+#Older GCC (<10) doesn't like -mabi=ieeelongdouble and -mlong-double-64.
+$(foreach suf,$(all-object-suffixes),\
+         $(objpfx)tst-nldbl-scanf-binary-c11$(suf) \
+         $(objpfx)tst-nldbl-scanf-binary-c2x$(suf) \
+         $(objpfx)tst-nldbl-scanf-binary-gnu11$(suf) \
+         $(objpfx)tst-nldbl-scanf-binary-gnu89$(suf) \
+         ): sysdep-CFLAGS := $(filter-out -mabi=ieeelongdouble,$(sysdep-CFLAGS))
+endif
+
+#Older GCC (<10) doesn't like -mabi=ieeelongdouble and -mlong-double-64.
+ifeq ($(subdir), wcsmbs)
+$(foreach suf,$(all-object-suffixes),\
+         $(objpfx)tst-nldbl-wscanf-binary-c11$(suf) \
+         $(objpfx)tst-nldbl-wscanf-binary-c2x$(suf) \
+         $(objpfx)tst-nldbl-wscanf-binary-gnu11$(suf) \
+         $(objpfx)tst-nldbl-wscanf-binary-gnu89$(suf) \
+         ): sysdep-CFLAGS := $(filter-out -mabi=ieeelongdouble,$(sysdep-CFLAGS))
 endif
 
 # Disable linker noise on files using ieee128 long double internally
